### PR TITLE
add licenses

### PR DESCRIPTION
**Pull request by ccoVeille**
- **feat: add Licenses**
- **feat: add use cases reported during code review**
- **fix: yaml errors**
- **fix BSD variations**
- **fix: Apache 2.0 version Licenses vs Apache2 webserser**

### CODE REVIEW

**Comment by ccoVeille** [comment]
This PR supersedes #39 merged too fast, and reverted with #40

**Comment by ccoVeille** [comment]
@alexvonme we can merge this one if you want

File: vale/styles/jargonLint/Licenses.yml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+---
+extends: substitution
+ignorecase: true
+level: warning
+message: Use '%s' instead of '%s'
+action:
+  name: replace
+swap:
+  # Apache[ -]?2(?:\.0)?: Apache 2.0 # also a webserver, to be fixed
+  "[0O]BSD": 0BSD
```
Comment (alexvonme) [inline vale/styles/jargonLint/Licenses.yml:10]: Is there a reason why this is between quotation while the others aren't.

Comment (ccoVeille) [reply vale/styles/jargonLint/Licenses.yml:10]: [] defines an array

so here we are writing

[0O] (an array containing 0O) and concat "BSD"

yaml is broken

```
vale/styles/jargonLint/Licenses.yml
  10:7      error    syntax error: expected <block end>, but found '<scalar>' (syntax)
```

Comment (ccoVeille) [reply vale/styles/jargonLint/Licenses.yml:10]: This is the reason I added yamllint to CI

- #43

Comment (alexvonme) [reply vale/styles/jargonLint/Licenses.yml:10]: I see, thanks for the explaination.

Comment (ccoVeille) [reply vale/styles/jargonLint/Licenses.yml:10]: E201 Invalid value [~/.local/share/vale/styles/jargonLint/Licenses.yml:9:1]:

   8  swap:
   9*   # Apache[ -]?2(?:\.0)?: Apache 2.0 # also a webserver, to be fixed
  10    [0O]BSD: 0BSD
  11    BSD: BSD   # also an operating system, but it's OK. BSD versions are not like BSD3 but 3BSD

did not find expected key

Here is what vale reports.

I also plan to validate the rules against vale directly, but yamllint is a no-brainer

**Comment by alexvonme** [comment]
@ccoVeille I significantly modified and built a methodology around a different Licenses.yml file which I propose in #46. The thing I disliked about the original was that it seems too similar to what we were trying to avoid, an unstructured and arbitrarily written rule.

Please don't take this as a rejection of your input. I'm trying to approach the same rule, with your input, through a verifiable methodology.

**Pull request closed without merging**